### ISSUE TITLE
Segmentation Fault with AUR bin

### ISSUE THREAD

**Issue by joe-herbert** [phuhl/linux_notification_center#218]
Firstly, love this project. It looks great and the customisability is fantastic.

Unfortunately, it often crashes on my system. I'm running Arch with BSPWM, with everything up to date and the latest version of `deadd-notification-center-bin` installed from the AUR.

When running `deadd-notification-center` it immediately fails with the following output about 75% of the time.
```
notificationDaemon started
deadd-notification-center: mmap 4096 bytes at (nil): Cannot allocate memory
deadd-notification-center: Try specifying an address with +RTS -xm<addr> -RTS
zsh: segmentation fault (core dumped)  deadd-notification-center
```

The remaining 25% of the time works great for a while, and then at some point fails with the same error. I haven't worked out what's triggering the failure as it seems to just happen randomly.

Is there a solution to this?

**Comment by phuhl** [comment]
Hi, thanks :)

Does your system have enough RAM? Looks like there is an issue with allocating more memory.

**Comment by ktarnaris** [comment]
I have the same problem. I have plenty of ram. Sometimes it is starting and crashes after a while

**Comment by phuhl** [comment]
Seems like a Haskell issue with the current kernel. Other reports:

https://github.com/IHaskell/IHaskell/issues/1396
https://bbs.archlinux.org/viewtopic.php?id=282429

You could check your kernel version with `uname -a` to confirm.

From the latter link:

> TL;DR It's a kernel bug. Not fixed in 6.1.10 (and most likely neither in 6.2.7).

and also

> The fix is queued for 6.2.9 [https://git.kernel.org/pub/scm/linux/ke … 1deb8d7db0](https://git.kernel.org/pub/scm/linux/kernel/git/stable/stable-queue.git/commit/queue-6.2/maple_tree-fix-mas_skip_node-end-slot-detection.patch?id=79583022efa64b59298c2cf4396bec1deb8d7db0)

Also, if you are comfortable to rebuild your kernel with a patch, that seems to do the trick for now.

>> If I understand correctly the fix is [https://git.kernel.org/pub/scm/linux/ke … 079a7ea480](https://git.kernel.org/pub/scm/linux/kernel/git/akpm/mm.git/commit/?h=mm-hotfixes-unstable&id=aa4ec634c24aa2ff24025ed1de763e079a7ea480) which is not in mainline yet.

> I applied this patch and it did fix the issue as reproduced on the GHC repo. Thanks for the link.

Othewise, we'll have to wait.

Closing as this is an upstream issue. Thanks for the reports, though.

**Comment by PLNech** [comment]
For the record, hitting the same error with kernel `6.3.2-arch1-1` :sweat:
